templates/create-daml-app-test-resources/index.test.ts

```diff
@@ -62,49 +54,35 @@ const removeFile = async (path: string) => {
 // Start the DAML and UI processes before the tests begin.
 // To reduce test times, we reuse the same processes between all the tests.
 // This means we need to use a different set of parties and a new browser page for each test.
-//
-// We run the sandbox and JSON API server separately, as opposed to using
-// `daml start` as we instruct users to do in the Getting Started Guide.
-// This is so that we have more control of the ports used and hopefully avoid
-// hardcoding port numbers (which can cause clashes in CI runs).
 beforeAll(async () => {
-  // If the sandbox or JSON API processes were previously shut down
-  // abruptly then their port files may not have been deleted.
-  // Since we use these files to know when the servers are up, we remove them
-  // first (if they exist) to be sure.
-  await removeFile(SANDBOX_PORT_FILE_PATH);
-  await removeFile(JSON_API_PORT_FILE_PATH);
-
-  // Run the `daml` commands from the project root (where the `daml.yaml` is located).
+  // If the JSON API server was previously shut down abruptly then the port file
+  // may not have been removed.
+  // Since we use this file to know when the server is up, we remove it first
+  // (if it exists) to be sure.
+  const jsonApiPortFilePath = `../${JSON_API_PORT_FILE_NAME}`; // relative to ui folder
+  await removeFile(jsonApiPortFilePath);
+
+  // Run `daml start` from the project root (where the `daml.yaml` is located).
   // The path should include '.daml/bin' in the environment where this is run,
   // which contains the `daml` assistant executable.
   const startOpts: SpawnOptions = { cwd: '..', stdio: 'inherit' };
 
-  // Port number 0 instructs the sandbox to find an available port and write the
-  // real port number to the port file.
-  const sandboxOptions = [
-    'sandbox',
-    '--wall-clock-time',
-    `--ledgerid=${SANDBOX_LEDGER_ID}`,
-    '--port=0',
-    `--port-file=${SANDBOX_PORT_FILE_NAME}`,
-    DAR_PATH
+  // Arguments for `daml start` (besides those in the `daml.yaml`).
+  // `--sandbox-port=0` instructs the sandbox to find an available port which is
+  // then used as the `--ledger-port` for the JSON API.
+  // The JSON API `--port-file` gives us a file we can check to know that both
+  // the sandbox and JSON API server are up and running.
+  // The JSON API port is hardcoded for now.
+  const startArgs = [
+    'start',
+    '--sandbox-port=0',
+    `--json-api-port=${JSON_API_PORT}`,
```

Comment:
I suggest we don't specify this port explicitly since the GSG uses the default. If we don't do the same here we're kind of blind against the GSG breaking because somebody changes the default port of JSON API and our tests don't break.